WalletWasabi.Tests/Helpers/HwiValidationHelper.cs

```diff
@@ -10,20 +11,52 @@ public static class HwiValidationHelper
 	/// </summary>
 	/// <param name="path">The wallet path which come from HWI enumerate command.</param>
 	/// <param name="model">The hardware wallet model.</param>
+	/// <param name="osPlatform">Mock test specific platform</param>
 	/// <returns><c>true</c> if the path matches the model's regex, <c>false</c> otherwise.</returns>
-	public static bool ValidatePathString(HardwareWalletModels model, string path)
+	public static bool ValidatePathString(HardwareWalletModels model, string path, OSPlatform? osPlatform = null)
```

Comment:
I don't think we need this new parameter anymore, it can be reverted. Same for the mocked tests.

Also, you can remove these extra checks `|| (osPlatform != null && osPlatform == OSPlatform.Windows)`. 
No need for them.

Reply:
Mock tests are only compatible with Windows operating system. You can check predefined paths. In that case, osplatform parameter is needed as the mocktest may fail.test will running to fail.

Reply:
We do not write code only for the tests, please remove this parameter, 
If the test is working only on Windows, then disable it on other platforms:

```csharp
if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
{
	return;
}
```

Here you are forcing the tests to act like they were running on Windows which is an extremely bad practice.

Reply:
The HwiKatas test is compatible with all platforms. However, the predefined data used in the mock test is designed specifically for Windows. 

I initially assumed that by passing Windows as a parameter in the mocktest, it would display the expected results, since this is only a mock test and not real data.